### PR TITLE
Support 'all' test levels in PR builds

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- PR regex has been updated to support 'all'
- Using 'all' will launch sanity and extended builds
- Can extend to more test levels when they are added

[skip ci]

Issue #1138

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

File: buildenv/jenkins/README.md [hunk 1]
```diff
@@ -41,10 +41,13 @@ This folder contains Jenkins pipeline scripts that are used in the OpenJ9 Jenkin
 - Request a Sanity build on zLinux all java versions
```
Comment (fjeremic) [inline buildenv/jenkins/README.md]: We should also change line 29:

>  - Current supported PR builds are "Compile", "Compile & Sanity", or "Compile & Extended"

Comment (AdamBrousseau) [reply buildenv/jenkins/README.md]: That statement is still true. I think we need another sentence that specifically calls out this feature.

**Comment by AdamBrousseau** [comment]
Commit amended @fjeremic

**Pull request merged**